### ISSUE TITLE
Out of memory when training with 1080ti

### ISSUE THREAD

**Issue by mariolew** [pradeeplam/Anime-Sketch-Coloring-with-Swish-Gated-Residual-UNet#12]
What GPU did you use?

**Comment by zamlz** [comment]
We trained on an Nvidia Titan XP GPU, but you should be able to train on a Nvidia 1080 Ti (my local machine uses one and was able to train) using a batch size of 1. This is also how the paper works as well. We tried experimenting with different batch sizes, but we couldn't get the loss function to work correctly and it was quite memory intensive.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by mariolew** [comment]
@zamlz  I use my own data and process with the providing code, the output image size is 512 512, however, the image size in the paper is 256 256

**Comment by zamlz** [comment]
The model will output the same size as the input image it was given. Our dataset was entirely of images of size 256x256. The model can be evaluated on any size after its been trained. But to train the model, using a 512x512 image dataset will cause memory problems for a 1080 Ti.

**Comment by mariolew** [comment]
Hi, I tried the pretrained model with some sketches from internet, the model performs much worse than style2paints.

**Comment by zamlz** [comment]
Yes, the approach is not very consistent with images outside of the training data. There are many possible reasons for this, but one possible reason is size of the image. Take a look at this test image I used,
![merged](https://user-images.githubusercontent.com/10750027/56782277-45ca7a00-679b-11e9-90c0-7cf22e32ef66.jpg)
The model's output is very bad. However, after downscaling the input sketch it became significantly better.
![merged2](https://user-images.githubusercontent.com/10750027/56782300-6266b200-679b-11e9-8fff-463aaee97a99.jpg)

There are also other limitations of the model. The sketches that were used to train the model were generated using [sketchKeras](https://github.com/lllyasviel/sketchKeras). In other words, the input data was generated using this other model to create realistic sketches (simply because we don't have enough real data of sketches to colored labels). This means our input data is biased as it expects that all sketches fall in this type of format. This means that real human made sketches may or may not fall into this same distribution of the input-data. 

Also the model doesn't work so well with manga as there is already enforced shading present, but surprisingly, on rare occasions, it still gets somewhat decent results.
![merged3](https://user-images.githubusercontent.com/10750027/56782541-79f26a80-679c-11e9-8e29-edf5d738041c.jpg)

**Comment by zamlz** [comment]
Here is another example that shows that the size of the features in the image really matter.
![merged](https://user-images.githubusercontent.com/10750027/56782620-baea7f00-679c-11e9-877c-0466f2301736.jpg)
![merged2](https://user-images.githubusercontent.com/10750027/56782621-baea7f00-679c-11e9-9e94-d279bdf739df.jpg)
Both are generated using the same pretrained weights that were uploaded.

**Comment by TachibanaYoshino** [comment]
Yes, due to the lack of a large number of manually drawn sketches and color anime images, the sketches currently used for training are extracted by other means. **And when this model is used for testing, the input sketch size should be 256 x 256, which is consistent with the resolution of the input sketch. It can be said that it is difficult to apply to input of other resolutions,** after all, **it is not a super-resolution AI.** Due to the limited storage space of a single GPU, the model is best adapted to a resolution of 256 x 256. You should write "**the resolution of the input sketch is 256 x 256 in the readme file, which will give you very good results."**
**If you want to color a manga, you should use the manga dataset for training.**

**Comment by TachibanaYoshino** [comment]
> 嗨，我试了一下预先训练过的模型和一些来自互联网的草图，模型的表现比风格2涂料差得多。

Please resize the size of the input sketch to 256 x 256 and attach the result image to see the coloring effect.

**Comment by TachibanaYoshino** [comment]
I tested it with some pictures. Here are the results of my test. The resolution of the input sketch is 256 x 256.
![image](https://user-images.githubusercontent.com/36946777/56845024-27749380-68ed-11e9-9e1a-66c9e8c9b66a.png)
![image](https://user-images.githubusercontent.com/36946777/56845026-2ba0b100-68ed-11e9-8848-831247173084.png)
![image](https://user-images.githubusercontent.com/36946777/56845027-2fccce80-68ed-11e9-87e2-0155b59b385a.png)
![image](https://user-images.githubusercontent.com/36946777/56845028-33605580-68ed-11e9-8039-5f103fc695a7.png)
![image](https://user-images.githubusercontent.com/36946777/56845032-38250980-68ed-11e9-87dc-e5396ed625d9.png)
![image](https://user-images.githubusercontent.com/36946777/56845035-3c512700-68ed-11e9-9717-8ed26e739a38.png)
![image](https://user-images.githubusercontent.com/36946777/56845039-43783500-68ed-11e9-82ad-c4db803319cc.png)
![image](https://user-images.githubusercontent.com/36946777/56845042-48d57f80-68ed-11e9-9556-043f1c7fdba7.png)
It looks pretty good.
